kernel/src/platform/mod.rs

```diff
@@ -101,6 +101,20 @@ pub trait Chip {
     /// chip.
     fn systick(&self) -> &Self::SysTick;
 
+    /// Tickles the watchdog timer if the Chip has one.
+    ///
+    /// This is called at the start of every kernel loop and when about to jump
+    /// to an app.
+    /// If a watchdog timer is supported, the board is expected to already have
+    /// enabled it with the correct frequency (using the `enable()` function in
+    /// the watchdog HIL). The interval must be at least as long as the
+    /// app time slice.
+    /// The chip must ensure that the watchdog timer is disabled while sleeping
+    /// and re-enabled afterwards.
```

Comment:
Shouldn't the kernel manage this, rather than each chip having to remember to do this?

Reply:
The disable and enable for sleep?

The `sleep()` is part of the `Chip` so right now it will be up to the `Chip`. We also don't currently expose a WDT to the kernel.

We could look at exposing the WDT to the kernel (like the MPU or systick) if that is a better option.